File: legacy/typescript/src/discord/doctor/data/DiscordInstallationDoctorResult.ts
export type DiscordDoctorStatus = 'PASS' | 'FAIL' | 'WARN'
export interface DiscordDoctorCheck { readonly key: string; readonly status: DiscordDoctorStatus; readonly message: string }
export interface DiscordInstallationDoctorResult { readonly healthy: boolean; readonly guildId: string; readonly guildName?: string; readonly applicationId: string; readonly botUserId?: string; readonly checks: readonly DiscordDoctorCheck[] }

File: src/main/java/org/tavall/community/doctor/DiscordInstallationDoctorResult.java
package org.tavall.community.doctor;

import java.util.List;

public record DiscordInstallationDoctorResult(
        boolean healthy,
        String guildId,
        String guildName,
        String applicationId,
        String botUserId,
        List<DiscordDoctorCheck> checks
) {
    public DiscordInstallationDoctorResult {
        guildId = requireText(guildId, "guildId");
        applicationId = requireText(applicationId, "applicationId");
        guildName = normalize(guildName);
        botUserId = normalize(botUserId);
        checks = List.copyOf(checks == null ? List.of() : checks);
    }

    private static String normalize(String value) {
        return value == null || value.isBlank() ? null : value.trim();
    }

    private static String requireText(String value, String fieldName) {
        if (value != null && !value.isBlank()) {
            return value.trim();
        }
        throw new IllegalArgumentException(fieldName + " must not be blank");
    }
}
